### PR TITLE
Some Linux kernel build fixes

### PR DESCRIPTION
**Pull request by Dennisbonke**
Mostly elf.h definitions.

### CODE REVIEW

File: options/elf/include/elf.h [hunk 1]
```diff
@@ -20,6 +20,7 @@ extern "C" {
 #define SHF_ALLOC 2
 #define SHF_EXECINSTR 4
 #define SHF_STRINGS 16
+#define SHF_INFO_LINK 32
```
Comment (no92) [inline options/elf/include/elf.h:23]: AFAICT this should be defined to 64, not 32.

Comment (no92) [reply options/elf/include/elf.h:23]: Oh and SHF_STRINGS should be 32, not 16.

Comment (no92) [reply options/elf/include/elf.h:23]: And SHF_TLS should be 1024.

**Review by no92 — CHANGES_REQUESTED** [review]

File: options/elf/include/elf.h (hunk 1 above)
Comment (Dennisbonke) [reply options/elf/include/elf.h]: Addressed as requested.

**Review by no92 — APPROVED** [review]
LGTM

**Review by Geertiebear — APPROVED** [review]
LGTM.

**Pull request merged**